commons/src/main/java/org/codice/ddf/admin/commons/requests/RequestUtils.java

```diff
@@ -0,0 +1,232 @@
+package org.codice.ddf.admin.commons.requests;
+
+import static java.net.HttpURLConnection.HTTP_OK;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.net.ssl.SSLPeerUnverifiedException;
+
+import org.apache.http.HttpRequest;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.ssl.SSLContexts;
+import org.apache.http.util.EntityUtils;
+import org.codice.ddf.admin.api.handler.ConfigurationMessage;
+import org.codice.ddf.admin.api.handler.MessageBuilder;
+import org.codice.ddf.admin.api.handler.report.ProbeReport;
+import org.codice.ddf.admin.api.handler.report.Report;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.net.HttpHeaders;
+
+public class RequestUtils {
+
+    //Probe return types
+    public static final String CONTENT_TYPE = "contentType";
+    public static final String CONTENT = "content";
+    public static final String STATUS_CODE = "statusCode";
+
+    //Success types
+    public static final String EXECUTED_REQUEST = "EXECUTED_REQUEST";
+    public static final String CONNECTED = "CONNECTED";
+    private static final Map<String, String> SUCCESS_DESCRIPTIONS = ImmutableMap.of(
+            CONNECTED, "Successfully established a connection with the endpoint.",
+            EXECUTED_REQUEST, "A request was successfully sent and a response was received from the endpoint.");
+
+    //Failure types
+    public static final String CERT_ERROR = "CERT_ERROR";
+    public static final String CANNOT_CONNECT = "CANNOT_CONNECT";
+    private static final Map<String, String> FAILURE_DESCRIPTIONS = ImmutableMap.of(
+            CANNOT_CONNECT, "The URL provided could not be reached.",
+            CERT_ERROR, "The discovered source has incorrectly configured SSL certificates and is insecure.");
+
+    //Warning types
+    public static final String UNTRUSTED_CA = "UNTRUSTED_CA";
+    public static final Map<String, String> WARNING_DESCRIPTIONS = ImmutableMap.of(UNTRUSTED_CA,
```

Comment:
You use the same message description - "The discovered URL has incorrectly configured SSL certificates and is likely insecure." - for both conditions.